src/command_modules/azure-cli-configure/azure/cli/command_modules/configure/_help.py

```diff
@@ -10,4 +10,12 @@
 helps['configure'] = """
             type: command
             short-summary: Configure Azure CLI 2.0 Preview or view your configuration. The command is interactive, so just type `az configure` and respond to the prompts.
+            examples:
+                - name: configure a default resource group
+                  text: >
+                    az configure --setting resource_group myRG
```

Comment:
IMO in the help, this should be `resource-group` instead of `resource_group`.
But in the actual config file, it should be saved in underscore casing.
So in your logic you can do `str.replace('-', '_')`. Then you could actually support both `-` and `_` with little effort.